app/src/main/kotlin/com/wire/android/ui/calling/initiating/InitiatingCallViewModel.kt

```diff
@@ -0,0 +1,84 @@
+package com.wire.android.ui.calling.initiating
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.wire.android.model.UserAvatarAsset
+import com.wire.android.navigation.EXTRA_CONVERSATION_ID
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.navigation.parseIntoQualifiedID
+import com.wire.kalium.logic.data.call.ConversationType
+import com.wire.kalium.logic.data.conversation.ConversationDetails
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.feature.call.usecase.EndCallUseCase
+import com.wire.kalium.logic.feature.call.usecase.StartCallUseCase
+import com.wire.kalium.logic.feature.conversation.ObserveConversationDetailsUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class InitiatingCallViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    private val navigationManager: NavigationManager,
+    private val startCall: StartCallUseCase,
+    private val endCall: EndCallUseCase,
+    private val conversationDetails: ObserveConversationDetailsUseCase
+) : ViewModel() {
+
+    var callInitiatedState by mutableStateOf(InitiatingCallState())
+        private set
+
+    val conversationId: QualifiedID = savedStateHandle
+        .get<String>(EXTRA_CONVERSATION_ID)!!
+        .parseIntoQualifiedID()
+
+    init {
+        viewModelScope.launch {
+            initiateCall()
+            initializeScreenState()
+        }
+    }
+
+    private suspend fun initializeScreenState() {
+        conversationDetails(conversationId = conversationId)
+            .collect {
+                callInitiatedState = when (it) {
+                    is ConversationDetails.Group -> {
+                        callInitiatedState.copy(
+                            conversationName = it.conversation.name,
+                            conversationType = ConversationType.OneOnOne
+                        )
+                    }
+                    is ConversationDetails.OneOne -> {
+                        callInitiatedState.copy(
+                            conversationName = it.otherUser.name,
+                            avatarAssetId = UserAvatarAsset(it.otherUser.completePicture ?: ""),
+                            conversationType = ConversationType.Conference
+                        )
+                    }
+                    else -> {
+                        callInitiatedState.copy(conversationName = null)
```

Comment:
what would be the "else" case here actually? what will happen if we set the conversation name to null ? do we really want to do that ?

Reply:
if the conversation name is null, we will display the "default" as a name..In same case, for example user removed from team, the name is returned  empty from backend

Reply:
maybe it will make more sense to add some sealed class to the conversation name that would have two states : 

Known, NotKnown or something like that where Known will propagate the name from the back-end and if there is not known you can pass the resource id like R.string.defaultName ? 

because for me it is a bit confusing why we are doing this here and we are kind of extending the logic to the UI where we check for the null of the conversationName so the logic just continues there

Reply:
But in this case, the UI will still hold the logic to check if we have the name or not, and use the corresponding sealed class